File: HandBallStats_api/src/main/java/es/diego/handballstats_api/models/Jugador.java
package es.diego.handballstats_api.models;

import es.diego.handballstats_api.models.estadisticas.Estadistica;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name ="jugador", uniqueConstraints = {@UniqueConstraint(columnNames = {"foto"})})
public class Jugador {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name="portero", nullable = false, columnDefinition = "BOOLEAN DEFAULT FALSE")
    private boolean portero = false;

    @Column(name = "nombre", nullable = false)
    private String nombre;

    @Column(name = "dorsal", nullable = false)
    private int dorsal;

    @Column(name = "foto")
    private String foto;

    @Column(name = "anio")
    private int anio;

    @ManyToOne
    @JoinColumn(name = "id_equipo", nullable = false)
    @JsonBackReference("b")
    private Equipo equipo;

    @OneToMany(mappedBy = "jugador", cascade = CascadeType.ALL)
    @JsonManagedReference("e")
    private List<Estadistica> estadisticas;
}

File: HandBallStats/app/src/main/java/es/diego/handballstats/models/Jugador.kt
package es.diego.handballstats.models

data class Jugador(
    var id: Int?,
    var portero: Boolean = false,
    var nombre: String,
    var dorsal: Int,
    var foto: String?,
    var anio: Int?,
    var equipo: Equipo?,
    var estadisticas: MutableList<Estadistica> = mutableListOf()
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Equipo) return false
        return this.id == other.id
    }

    override fun hashCode(): Int {
        return id ?: 0
    }
}